ATOMS.md

```diff
@@ -0,0 +1,63 @@
+## Mobiledoc Atoms
+
+Atoms are effectively read-only inline cards.
+
+An atom is an object with a name and at least one of several hooks defined. For example:
+
+```js
+var demoAtom = {
+  name: 'mention',
+  display: {
+    setup(element, options, env, text, payload) {
+      element.innerHTML = `<span class="mention">${text}</span>`;
+    }
+  }
+}
+```
+
+### Available hooks
+
+Like [CARDS](CARDS.md) there are several hooks an atom should define.
+
+```js
+var exampleAtom = {
+  name: 'example',
+  display: {
+    setup(element, options, env, text, payload) {},
+    teardown(setupReturnValue) {}
+  },
+  html: {
+    setup(buffer, options, env, text, payload) {},
+    teardown(setupReturnValue) {}
+  }
+};
+```
+
+|Hook|Used by Mobiledoc Kit|Used by DOM Renderer|Used by HTML Renderer|
+|---|---|---|---|
+|`display`|✓|✓||
+|`html`|||✓|
+
+Atoms are read-only so, unlike cards, do not have an `edit` hook.
+
+Each hook has a `setup` and `teardown` method. The arguments are:
+
+* `element` is a DOM element for that section. Nodes of that view for a atom
+  should be appended to the `element`.
+* `buffer` is an array passed to the `html` hook instead of a DOM element.
+  The content for the atom should be pushed on that array as a string.
+* `options` is the `cardOptions` argument passed to the editor or renderer.
+* `env` contains information about the running of this hook. It may contain
```

Comment:
If a card can remove itself, can it also save itself? It if can trigger its own destruction, it seems reasonable that it could change its payload as well.

On the other hand, part of me thinks it should not be able to remove or edit itself. The most conservative approach is that they can only be created and if they are removed it must be through normal editing operations like backspace and delete. Thoughts?